pyomo/contrib/mindtpy/outer_approximation.py

```diff
@@ -0,0 +1,445 @@
+                                self.working_model)
+                            if self.curr_int_sol in set(self.integer_list):
+                                config.logger.info(
+                                    'The same combination has been explored and will be skipped here.')
+                                continue
+                            else:
+                                self.integer_list.append(
+                                    self.curr_int_sol)
+                        counter += 1
+                        fixed_nlp, fixed_nlp_result = self.solve_subproblem(config)
+                        self.handle_nlp_subproblem_tc(fixed_nlp, fixed_nlp_result, config)
+
+                        # Call the NLP post-solve callback
+                        with time_code(self.timing, 'Call after subproblem solve'):
+                            config.call_after_subproblem_solve(fixed_nlp)
+
+                        if self.algorithm_should_terminate(config, check_cycling=False):
+                            last_iter_cuts = True
+                            break
+
+                        if counter >= config.num_solution_iteration:
+                            break
+
+            # if config.strategy == 'PSC':
+            #     # If the hybrid algorithm is not making progress, switch to OA.
+            #     progress_required = 1E-6
+            #     if solve_data.objective_sense == minimize:
+            #         log = solve_data.LB_progress
+            #         sign_adjust = 1
+            #     else:
+            #         log = solve_data.UB_progress
+            #         sign_adjust = -1
+            #     # Maximum number of iterations in which the lower (optimistic)
+            #     # bound does not improve before switching to OA
+            #     max_nonimprove_iter = 5
+            #     making_progress = True
+            #     # TODO-romeo Unnecessary for OA and ROA, right?
+            #     for i in range(1, max_nonimprove_iter + 1):
+            #         try:
+            #             if (sign_adjust * log[-i]
+            #                     <= (log[-i - 1] + progress_required)
+            #                     * sign_adjust):
+            #                 making_progress = False
+            #             else:
+            #                 making_progress = True
+            #                 break
+            #         except IndexError:
+            #             # Not enough history yet, keep going.
+            #             making_progress = True
+            #             break
+            #     if not making_progress and (
+            #             config.strategy == 'hPSC' or
+            #             config.strategy == 'PSC'):
+            #         config.logger.info(
+            #             'Not making enough progress for {} iterations. '
+            #             'Switching to OA.'.format(max_nonimprove_iter))
+            #         config.strategy = 'OA'
```

Comment:
Do you need these comments?

Reply:
These might be useful for later PSC implementation.

Reply:
We rather leave these in for the moment unless you have a strong reason to remove them.

Reply:
Leaving them is fine. (They will of course be in the commit history even if you remove them, but that doesn't always make them easy to find...)